flink-core/src/main/java/org/apache/flink/core/memory/HybridMemorySegment.java

```diff
@@ -357,4 +385,9 @@ public final void put(int offset, ByteBuffer source, int numBytes) {
             }
         }
     }
+
+    @Override
+    public <T> T processAsByteBuffer(Function<ByteBuffer, T> processFunction) {
+        return Preconditions.checkNotNull(processFunction).apply(wrapInternal(0, size));
+    }
```

Comment:
Which calls to `wrap` cannot be replaced with this call?

Reply:
Do we know what the performance penalty for the additional `Function` is?

Reply:
> Which calls to wrap cannot be replaced with this call?

Technically, I don't think there are `wrap` calls that CANNOT be replaced. Some of the calls only wrap part of the segment, which can be solved by simply adding `offset` and `length` as arguments. I have not replaced all calls to `wrap` with this `processAsByteBuffer`, mostly because the remaining calls are on direct segment, thus it's not necessary to migrate them.

> Do we know what the performance penalty for the additional Function is?

We don't have a measurement. However, given that this method is for the per-segment operation (by default 32KB), I don't expect a significant performance penalty from using the `Function`.

Reply:
We could think about replacing these calls with `processAsByteBuffer` also for the direct memory in order to be able to get rid of the `wrap` method.

Reply:
I'll look into them together with the follow-up refactor effort. Since not all `wrap` calls are on the entire segment, I'll need to check the performance impact from using `Function` on those calls.

Reply:
Maybe it is also ok to provide additional methods on the `MemorySegment` which use internally a `ByteBuffer` implementation to be efficient.